modules/setting/session.go

```diff
@@ -50,7 +50,7 @@ func loadSessionFrom(rootCfg ConfigProvider) {
 	}
 	SessionConfig.CookieName = sec.Key("COOKIE_NAME").MustString("i_like_gitea")
 	SessionConfig.CookiePath = AppSubURL + "/" // there was a bug, old code only set CookePath=AppSubURL, no trailing slash
-	SessionConfig.Secure = sec.Key("COOKIE_SECURE").MustBool(false)
+	SessionConfig.Secure = sec.Key("COOKIE_SECURE").MustBool(strings.HasPrefix(strings.ToLower(AppURL), "https://"))
```

Comment:
On second thought, I would actually drop reading setting key at all and remove it from configs and leaving just this assignment based on app url. From security point of view there is no point in making it use unsecured cookie over https

Reply:
See #26953

Theoretically yes. But: 

1. There is no "dynamic cookie secure" support in the go-chi package (it could be improved separately)
2. "dynamic cookie secure" will leave security problems if the site admin doesn't configure the X-Forwarded-Proto correctly, because Gitea self might be running in HTTP mode behind an HTTPS proxy.

Reply:
points 1 and 2 are not really relevant for this solution but as for 3 I don't really see such use-case so in name of security I would propose to drop this setting in future

Reply:
Sure, the change has been backport to 1.20, let's break it for 1.21 later

Reply:
> points 1 and 2 are not really relevant for this solution but as for 3 I don't really see such use-case so in name of security I would propose to drop this setting in future

The case: https://github.com/go-gitea/gitea/issues/29451